modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/utilities/NetCDFUtilities.java

```diff
@@ -636,6 +659,93 @@ public static boolean isVariableAccepted(final String name,
             return true;
         }
     }
+        
+    public static FileFormat getFormat(URI uri) throws IOException {
```

Comment:
Note: this is refactored with the existing code pulled out into a method and some XML handling logic added.